src/main/java/org/carlmontrobotics/subsystems/IntakeShooter.java

```diff
@@ -1,50 +1,161 @@
+        //Take into note that in reality, the 2 points can return 2 possible centers
+        return k - center; //<- offset from the center
+        // double a = 4 * Math.pow(distanceBetweenSensors,2); <- for speaker calcs
+        // return 1.2;
+    }
+
+    public double calculateIntakeAmount(){
+        //Literatly just calcDIstanceSensorNotes but instead of solving for k, we are solving for h
+        double d1 = getGamePieceDistance1();
+        double d2 = getGamePieceDistance2();
+        double r = 7;
+
+        double xm = (distanceBetweenSensors)/2;
+
+        double h = xm + (Math.sqrt(Math.pow(r,2) - Math.pow(r/2,2)) * (d1-d2))/r; 
+        return h;
     }
 
-  
-    
     @Override
     public void periodic() {
-			//is pid needed? for rpm speeds?
-		}
+        SmartDashboard.putNumber("Outake Velocity", outakeEncoder.getVelocity());
+        SmartDashboard.putNumber("Intake Velocity", intakeEncoder.getVelocity());
+        SmartDashboard.putNumber("distance sensor 1", getGamePieceDistance1());
+        SmartDashboard.putNumber("distance sensor 2", getGamePieceDistance2());
+        SmartDashboard.putBoolean("DS1 Sees piece", gameDistanceSees1st());
+        SmartDashboard.putBoolean("DS2 Sees piece", gameDistanceSees2nd());
+        senseGamePieceStop();
+    }
+
+    public void setRPMOutake(double rpm) {
+        pidControllerOutake.setReference(rpm, CANSparkBase.ControlType.kVelocity, 0, feedforward.calculate(rpm));
+    }
+
+    public void setRPMintake(double rpm) {
+        pidControllerIntake.setReference(rpm, CANSparkBase.ControlType.kVelocity, 0, feedforward.calculate(rpm));
+    }
+
+
+
+    public void shoot(double distance) {
+        double rpm = calculateDistanceForRPM();
+        pidControllerOutake.setReference(rpm, CANSparkBase.ControlType.kVelocity, 0, feedforward.calculate(rpm));
+        //TODO: To be implemented
+    }
+
+    public void stopOutake() {
+        pidControllerOutake.setReference(0, CANSparkBase.ControlType.kVelocity, 0, feedforward.calculate(0));
+    }
+
+    public void stopIntake() {
+        pidControllerIntake.setReference(0, CANSparkBase.ControlType.kVelocity, 0, feedforward.calculate(0));
+    }
+    public void setRPMEjectOutake() {
+        pidControllerOutake.setReference(-3000, CANSparkBase.ControlType.kVelocity, 0, feedforward.calculate(-3000));
+    }
+    public void setRPMEjectIntake() {
+        pidControllerOutake.setReference(3000, CANSparkBase.ControlType.kVelocity, 0, feedforward.calculate(3000));
+    }
+
+    public double calculateDistanceForRPM() {
+        //returns specific rpm based off of the distance and angle it is in
+        double distance = 30; //This will be the x value returned from lime light
+        
+        double SpeakerHeight = 40; //Use limelight to return y for Speaker Height  Not a constant since the height will change when farther away 
```

Comment:
these should be in constants, and distance should be an input of the function, not a hardcoded number.